src/lib/list/list.scss

```diff
@@ -60,18 +73,18 @@ $mat-dense-list-icon-size: 20px;
     height: $avatar-height;
   }
 
-  &.mat-2-line .mat-list-item-content {
+  &.mat-2-line .mat-list-item-content .mat-list-item-content-reverse {
```

Comment:
You shouldn't need to add `mat-list-item-content-reverse` to these existing styles; the `mat-list-item-content-reverse` class should only have the extra styles you need to reverse the display